### ISSUE TITLE
Simultaneous turns multiplayer game: UI not properly locking when first player presses end turn button

### ISSUE THREAD

**Issue by dydzio0614** [vcmi/vcmi#3436]
**Describe the bug**
When you press end turn button as first player during simultaneous turns... it is possible to press end turn button multiple times, open hero window, town window etc. - UI is not properly locked until you perform some UI change that will properly trigger the lock. Not sure about exact constraints required (host/non-host, red/non-red color etc.) but this can happen. I think it is also technically possible to build building (and possibly do other stuff) at that time if you did not build it on your turn.

**To Reproduce**
1. Play multiplayer game with simultanoeus turns still in effect
2. End turn before opponent does
3. You should see end turn button not going disabled instantly after first press and you can still double click your current hero etc. to open his details - maybe needs additional conditions to trigger (not being host, not being red etc.)

**Expected behavior**
Either consistently lock UI or implement proper browsing of heroes/towns during human opponent turn - not sure about AI turn in multiplayer games with AI but many templates are played without AI

**Actual behavior**
You can open heroes/towns and see unblocked build building in towns etc. until some UI action properly locks UI

**Did it work earlier?**
Simultaneous turns are new feature, so not applicable to earlier versions

**Version**
 - OS: Kubuntu 22.04
 - Version: 1.4.2

**Additional context**
This bug is also semi-feature since in the end we want to get possibility of browsing heroes and towns during opponent turn, but performing actions such as building buildings, swapping artifacts etc. should be always impossible

**Comment by IvanSavenko** [comment]
UI will now be locked as expected.

Allowing actions during enemy turns makes sense, but is a separate issue and should be allowed even outside of simturns
